core/templates/dev/head/pages/collection_player/CollectionPlayer.js

```diff
@@ -283,6 +283,17 @@ oppia.controller('CollectionPlayer', [
       $location.hash(id);
       $anchorScroll();
     };
+
+    $scope.closeOnClickingOutside = function($evt) {
```

Comment:
I don't think you should be doing anything with $evt here. Couldn't you just toggle/close the preview card programmatically?

I guess you're trying to make it so that the card isn't closed when you click on the card, but you should do that using CSS to have the card have a higher z-index than the background layer, or something. Otherwise what you have here will break if the DOM structure subsequently changes  -- it's not maintainable.

Reply:
@seanlip All the solution I have read on StackOverflow and elsewhere make use of $evt to close a modal on clicking outside.
And I don't think click event has something to do with z-index or any other CSS property. I agree it is not such a decent solution but it will work until nobody enters another div between exploration card and background mask div.

Reply:
That's the problem -- you can't assume that the div structure won't change.

However, I'm not sure I understand the issue. Why can't you have a background mask div that has an ng-click, and have the card be positioned "above" it on the page, such that when the card is clicked you block the event from propagating further to the background (and when anywhere else is clicked, the ng-click on the background mask div fires)?

Reply:
I misunderstood what you said above.

Reply:
Done